testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -1295,6 +1393,41 @@ fn microblock_integration_test() {
             .unwrap()
     );
 
+    // wait at least two p2p refreshes so it can produce the microblock
+    for i in 0..30 {
```

Comment:
You should be able to use the `PoxWatchdog` to poll for this, so you don't have to worry about races.

Reply:
On second thought, plumbing this through would probably be more effort than it's worth.

Reply:
Discussed in discord that this change isn't strictly necessary.